src/Components/Aspire.Azure.Messaging.WebPubSub/AspireWebPubSubExtensions.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Azure.Common;
+using Aspire.Azure.Messaging.WebPubSub;
+
+using Azure;
+using Azure.Core;
+using Azure.Core.Extensions;
+using Azure.Messaging.WebPubSub;
+
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering <see cref="WebPubSubServiceClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireWebPubSubExtensions
+{
+    private const string DefaultConfigSectionName = "Aspire:Azure:Messaging:WebPubSub";
+
+    /// <summary>
+    /// Registers <see cref="WebPubSubServiceClient"/> as a singleton in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="hubName">The hub used for Web PubSub service.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingWebPubSubSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{WebPubSubServiceClient, WebPubSubServiceClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.Messaging.WebPubSub" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingWebPubSubSettings.ConnectionString"/> nor <see cref="AzureMessagingWebPubSubSettings.Endpoint"/> is provided.</exception>
+    public static void AddAzureWebPubSub(
```

Comment:
```suggestion
    public static void AddAzureWebPubSubClient(
```

We added `Client` to all our extension methods.

Reply:
WebPubSub is a little bit different than other service's Rest Client, as WebPubSub has another real client role, as shown in this diagram: 

![image](https://github.com/dotnet/aspire/assets/668244/b1b44fa0-b156-4ab3-8019-2364ac6f1707)

And we also have 2 packages, one is [Azure.Messaging.WebPubSub](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/webpubsub/Azure.Messaging.WebPubSub) which aspire component adds, another is [Azure.Messaging.WebPubSub.Client](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/webpubsub/Azure.Messaging.WebPubSub.Client) which is for the real client role. 
`AddAzureWebPubSubClient` might be a bit confusing for WebPubSub, maybe `AddAzureWebPubSubServiceClient`? I actually prefer the shorter one `AddAzureWebPubSub` ....

Reply:
The reason we changed the method names is to disambiguate the methods between the AppHost and the apps:

````
```csharp
var webPubSub = builder.AddAzureWebPubSub("wps");

var myService = builder.AddProject<Projects.MyService>()
                       .WithReference(webPubSub);
```

The `AddAzureWebPubSub` method will read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:wps` config key. The `WithReference` method passes that connection information into a connection string named `wps` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:

```csharp
builder.AddAzureWebPubSub("wps", "your_hub_name");
```
````

Notice both methods at the top and bottom are `builder.AddAzureWebPubSub`. We received feedback that this is confusing to name the methods the same.

So to be consistent with the rest of the components, we should rename this method.

Reply:
> maybe AddAzureWebPubSubServiceClient?

I'd be fine with that.

Reply:
another option is `AddAzureWebPubSubHub` since the method also requires `hub` parameter, how do you feel about this one @eerhardt ? I am a little concerned about appending `ServiceClient` since that would limit this method to add `ServiceClient`, there is an aspnetcore middleware https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore that I'd like to add when later the ideal workflow https://github.com/dotnet/aspire/pull/2495#discussion_r1506957056 is possible. This middleware is to help handle the traffic in the upstream server from the client.

Reply:
> another option is AddAzureWebPubSubHub since the method also requires hub parameter, how do you feel about this one @eerhardt ?

I'm fine with appending "hub" if that's what you feel is best. Note that other components use the term "Client" and don't necessarily mean the DI service is named "Client". For example, RabbitMQ, NATS, and Redis all have a "connection" object, but their extension methods are named "AddXXXClient", meaning this app is going to be a "client" of the XXX server.

The main point here is that we don't want the same names between the AppHost extension method and the Component extension method.

> there is an aspnetcore middleware

Note that the middleware can't go in this method because it is too early in the app's lifecycle. Middleware get added after the `builder` is "built" into an app, and are added to the "app" object.

Reply:
Changed to ...Hub considering WebPubSub plays more like a server role.

> Note that the middleware can't go in this method because it is too early in the app's lifecycle. Middleware get added after the builder is "built" into an app, and are added to the "app" object.

Understood. `AddAzureWebPubSubHub` would add necessary dependencies, and users can call `app.UseWebPubSubHub` in their middleware if `builder.AddAzureWebPubSubHub`.